src/main/webapp/src/app/create-gardens-form/create-gardens.component.ts

```diff
@@ -22,5 +29,64 @@ import {Component, OnInit} from '@angular/core';
   ],
 })
 export class CreateGardensComponent implements OnInit {
-  ngOnInit() {}
+  @Input('email') adminEmail: string;
+  gardenName: string;
+  gardenDescription: string;
+  gardenZip: string;
+  done = false;
+
+  gardenProfile: Garden = {
+    id: undefined,
+    name: undefined,
+    description: undefined,
+    lat: undefined,
+    lng: undefined,
+    zipCode: undefined,
+    adminId: undefined,
+  };
+
+  constructor(private httpClient: HttpClient, private router: Router) {}
+
+  gardenGroup: FormGroup;
+
+  ngOnInit(): void {
+    this.gardenGroup = new FormGroup({
+      gardenNameValidator: new FormControl(this.gardenName, [
+        Validators.required,
+      ]),
+      gardenDescriptionValidator: new FormControl(this.gardenDescription, [
+        Validators.required,
+      ]),
+      gardenZipValidator: new FormControl(this.gardenZip, [
+        Validators.required,
+      ]),
+    });
+  }
+
+  buildGardenProfile(): void {
+    this.gardenProfile = {
+      id: '2',
+      name: this.gardenName,
+      description: this.gardenDescription,
+      lat: 0.0,
+      lng: 0.0,
+      zipCode: this.gardenZip,
+      adminId: this.adminEmail,
+    };
+    this.done = true;
+    this.postGardenData(this.gardenProfile);
+  }
+
+  postGardenData(data: Garden): void {
+    const httpOptions = {
+      params: new HttpParams().set('gardenData', JSON.stringify(data)), //.set('isIdValid', this.isValidID.toString()),
```

Comment:
In general it's not good practice to leave code commented out in a codebase. It's a slippery slope and can get messy quick! I recommend if you can remove entirely or instead leave a TODO with an issue number to restore something, that can be a more manageable way to maintain outstanding tasks.